tools/android/json_worker_wrapper.py

```diff
@@ -0,0 +1,55 @@
+from io import StringIO
+import json
+import select
+import os
+import sys
+import traceback
+
+
+def wait_for_stdin_json():
+  stdin = open("/dev/stdin", "rb")
+  os.set_blocking(stdin.fileno(), False)
+  stream_bytes = b""
+  while True:
+    _, _, _ = select.select([stdin.fileno()], [], [])
+    maybe_bytes = stdin.read()
+    if maybe_bytes is None:
+      continue
+    stream_bytes += maybe_bytes
+    try:
+      work_request = json.loads(stream_bytes)
+    except json.decoder.JSONDecodeError:
+      # stream couldn't decode, wait until next select
+      sys.stderr.write("failed to decode: " + stream_bytes.decode('utf-8') + "\n")
+      sys.stderr.flush()
```

Comment:
Manually manipulating sys.stderr seems like a brittle way to do this. Are there no python builtin convenience functions to abstract some of this detail away?